core/src/main/java/feign/ParamAlias.java

```diff
@@ -0,0 +1,20 @@
+package feign;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Allows specify query parameter name different from field name, when Object is used like parameters container
+ *
+ * @see feign.querymap.BeanQueryMapEncoder
+ * @see feign.querymap.FieldQueryMapEncoder
+ */
+@Retention(RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface ParamAlias {
```

Comment:
Any particular reason no to use existing `@Param`

Reply:
That would be my recommendation as well.  Consider using our existing `@Param` annotation.